laspy/lasreader.py

```diff
@@ -151,48 +142,11 @@ def read(self) -> LasData:
                 # In that case we are still going to
                 # try to read the evlrs by relying on the fact that they should generally be
                 # right after the last point, which is where we are now.
-                if self.header.are_points_compressed:
-                    if not isinstance(self.point_source, LazrsPointReader):
-                        raise errors.LaspyException(
-                            "Reading EVLRs from a LAZ in a non-seekable stream "
-                            "can only be done with lazrs backend"
-                        )
-                    # Few things: If the stream is non seekable, only a LazrsPointReader
-                    # could have been created (parallel requires ability to seek)
-                    #
-                    # Also, to work, the next lines of code assumes that:
-                    # 1) We actually are just after the last point
-                    # 2) The chunk table _starts_ just after the last point
-                    # 3) The first EVLR starts just after the chunk table
-                    # These assumptions should be fine for most of the cases
-                    # and non seekable sources are probably not that common
-                    _ = self.point_source.read_chunk_table_only()
-
-                    # Since the LazrsDecompressor uses a buffered reader
-                    # the python file object's position is not at the position we
-                    # think it is.
-                    # So we have to read data from the decompressor's
-                    # buffered stream.
-                    class LocalReader:
-                        def __init__(self, source: LazrsPointReader) -> None:
-                            self.source = source
-
-                        def read(self, n: int) -> bytes:
-                            return self.source.read_raw_bytes(n)
-
-                    self.evlrs = VLRList.read_from(
-                        LocalReader(self.point_source),
-                        self.header.number_of_evlrs,
-                        extended=True,
-                    )
-                else:
-                    # For this to work, we assume that the first evlr
```

Comment:
Although it couples the LasReader with the laz backend implementation which should be interacted in an abstract way, I still want to keep the code as it is for a few reasons:

- I feel like refactoring this is out of the scope of the PR
- I don't feel like the refactor makes the code simpler to understand, I would even argue that it makes it harder/impossible to understand:

In the LazrsPointReader you added 
```Python

   def read(self, n: int) -> bytes:
        if not self._started_reading:
            self.read_chunk_table_only()
            self._started_reading = True
        return self.read_raw_bytes(n)
```

so that the line 
```Python
self.evlrs = VLRList.read_from(
                    self.point_source.source,
                    self.header.number_of_evlrs,
                    extended=True,
                )
```

produces the same behaviour as before the refactor.

However this has 2 problems:
1) When reading the implemetation for LazrePointReader.read, I have no idea why this read_chunk_table only is needed, and why only if we haven't started reading. 99% sure, than in 6 months we won't understand why its done this way, what's the porpuse of the code. 
Whereas I would argue that the original code is clearer thanks to the fact that it is isolated to where its needed + heavily commented.

2) The implementation of `read` which is a general function, now is implemented to work around a very very specific case. And so this read method only works for this very very specific case.

Meaning someone could call the read function but things would go bad as it might not call it in the only case where this function is useful.

Reply:
After reading this part anew, I agree that the workaround inside the `read` function is just wrong. This part will be reworked.